### PR TITLE
DocumentCard: added native properties to the root element

### PR DESCRIPTION
**Pull request by mtennoe**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

The DocumentCard was lacking the ability to override the aria label and listening to onMouseLeave and onMouseEnter events. This adds those and other native properties to the root element of the documentcard

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by mikewheaton** [comment]
Looks like the build should succeed after you run `npm run update-api` and commit the changes.

**Comment by Markionium** [comment]
@mtennoe it might make sense to use `getNativeProps()` it would probably solve all 3 props + many others.

See this file for an example
https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Spinner/Spinner.base.tsx#L17

File: packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ export class DocumentCardBase extends BaseComponent<IDocumentCardProps, any> imp
 
   public render(): JSX.Element {
     const { onClick, onClickHref, children, type, accentColor, styles, theme, className } = this.props;
+    const nativeProps = getNativeProps(this.props, divProperties, ['onClick', 'type']);
```
Comment (mtennoe) [inline packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx]: @dzearing - If I didn't exclude the "type" property the snapshot tests would suddenly start including a type={0} in the resulting snapshot. Do you know if this is a bug with getNativeProps? It seemed unexpected, so Mark and I decided to exclude the property to be on the safe side

Comment (mtennoe) [reply packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.base.tsx]: We figured it out, we didnt notice the "type" property which was already defined in the properties type

**Pull request merged**